pkg/kvstore/etcd.go

```diff
@@ -568,7 +568,6 @@ func (e *etcdClient) renewSession(ctx context.Context) error {
 	//  - The above call to sessionCancel() cancels the session due to the
 	//  session ending and requiring renewal.
 	sessionContext, sessionCancel, sessionSuccess := contexthelpers.NewConditionalTimeoutContext(ctx, statusCheckTimeout)
```

Comment:
I'm not sure if I agree with this commit. The GC will only happen after `statusCheckTimeout`, if the `statusCheckTimeout` is 10 hours we could have stopped the go routine earlier by running `defer close(sessionSuccess)`

Reply:
I didn't think about long timeouts. OK, let me drop that commit.